yt_dlp/extractor/booyah.py

```diff
@@ -0,0 +1,91 @@
+from .common import InfoExtractor
+from ..utils import int_or_none, str_or_none, traverse_obj
+
+
+class BooyahBaseIE(InfoExtractor):
+    _BOOYAH_SESSION_KEY = None
+
+    def _real_initialize(self):
+        if not BooyahBaseIE._BOOYAH_SESSION_KEY:
+            initial_request = self._request_webpage(
+                'https://booyah.live/api/v3/auths/sessions', None, data="".encode())
+            BooyahBaseIE._BOOYAH_SESSION_KEY = initial_request.getheader('booyah-session-key')
+
+
+class BooyahClipsIE(BooyahBaseIE):
+    _VALID_URL = r'https://booyah.live/clips/(?P<id>\d+)'
+    _TESTS = [{
+        'url': 'https://booyah.live/clips/13887261322952306617',
+        'info_dict': {
+            'id': '13887261322952306617',
+            'ext': 'mp4',
+            'view_count': int,
+            'duration': 30,
+            'channel_id': 90565760,
+            'like_count': int,
+            'title': 'Cayendo con estilo 😎',
+            'uploader': '♡LɪꜱGΛ​MER​',
+            'comment_count': int,
+            'uploader_id': '90565760',
+            'thumbnail': 'https://resmambet-a.akamaihd.net/mambet-storage/Clip/90565760/90565760-27204374-fba0-409d-9d7b-63a48b5c0e75.jpg',
+            'upload_date': '20220617',
+            'timestamp': 1655490556,
+            'modified_timestamp': 1655490556,
+            'modified_date': '20220617',
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        json_data = self._download_json(
+            f'https://booyah.live/api/v3/playbacks/{video_id}', video_id,
+            headers={'Booyah-Session-key': BooyahBaseIE._BOOYAH_SESSION_KEY})
+
+        formats = []
+        for video_data in json_data['playback']['endpoint_list']:
+            fmt = {
+                'url': video_data.get('stream_url'),
+                'ext': 'mp4',
+                'resolution': f'{video_data.get("resolution")}p' if video_data else None
+            }
+            watermark_fmt = {
+                'url': video_data.get('download_url'),
+                'ext': 'mp4',
+                'format_note': 'Watermarked',
+                'resolution': f'{video_data.get("resolution")}p' if video_data else None
+            }
+            formats.append(fmt)
+            formats.append(watermark_fmt)
```

Comment:
You can inline this to avoid the variables

Also I think its better to set height than resolution, it will add the "p" automagically

```python
'height': video.get('resolution')
```